libs/dh/admin/feature-user-roles/src/lib/download.component.ts

```diff
@@ -35,30 +47,60 @@ import { DhUserRoles } from '@energinet-datahub/dh/admin/data-access-api';
   }}</watt-button>`,
 })
 export class DhUserRolesDownloadComponent {
+  private environment = inject(dhAppEnvironmentToken);
+  private toastService = inject(WattToastService);
   private transloco = inject(TranslocoService);
+  private query = lazyQuery(GetUserRolesForCsvDocument);
 
-  userRoles = input.required<DhUserRoles>();
+  filters = input.required<Variables>();
 
   async download() {
-    this.transloco
-      .selectTranslateObject('marketParticipant.marketRoles')
-      .pipe(take(1))
-      .subscribe((rolesTranslations) => {
-        const basePath = 'admin.userManagement.tabs.roles.table.columns.';
+    this.toastService.open({
+      type: 'loading',
+      message: translate('shared.downloadStart'),
+    });
+
+    try {
+      const result = await this.query.query({
+        variables: {
+          ...this.filters(),
+          first: 10_000,
+          order: {
+            name: SortEnumType.Asc,
+          },
+        },
+      });
+
+      const userRoles = result.data.filteredUserRoles?.nodes ?? [];
+
+      this.transloco
+        .selectTranslateObject('marketParticipant.marketRoles')
+        .pipe(take(1))
+        .subscribe((rolesTranslations) => {
+          const basePath = 'admin.userManagement.tabs.roles.table.columns.';
 
-        const headers = [
-          `"${translate(basePath + 'name')}"`,
-          `"${translate(basePath + 'marketRole')}"`,
-          `"${translate(basePath + 'status')}"`,
-        ];
+          const headers = [
+            `"${translate(basePath + 'name')}"`,
+            `"${translate(basePath + 'marketRole')}"`,
+            `"${translate(basePath + 'status')}"`,
+          ];
 
-        const lines = this.userRoles().map((role) => [
-          `"${role.name}"`,
-          `"${rolesTranslations[role.eicFunction]}"`,
-          `"${translate('admin.userManagement.roleStatus.' + role.status)}"`,
-        ]);
+          const lines = userRoles.map((role) => [
+            `"${role.name}"`,
+            `"${rolesTranslations[role.eicFunction]}"`,
+            `"${translate('admin.userManagement.roleStatus.' + role.status)}"`,
+          ]);
+          const env = translate(`envinronementName.${this.environment.current}`);
+          const fileName = `DataHub-User-roles-${wattFormatDate(new Date(), 'long')}-${env}`;
```

Comment:
praise: nice work for making the file name more descriptive. The user's download uses something similar which was appreciated by our service desk.

suggestion: it's a small details but it would be awesome to unify the download file names (where it makes sense). It will make the experience more unified. Let's track it in a separate task (will create one on our board)